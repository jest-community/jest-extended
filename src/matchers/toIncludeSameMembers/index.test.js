import matcher from './';

expect.extend(matcher);

describe('.toIncludeSameMembers', () => {
  test('passes when arrays are empty', () => {
    expect([]).toIncludeSameMembers([]);
    expect(new Set()).toIncludeSameMembers([]);
  });

  test('passes when arrays match', () => {
    expect([1, 2, 3]).toIncludeSameMembers([1, 2, 3]);
    expect(new Set([1, 2, 3])).toIncludeSameMembers([1, 2, 3]);
    expect([{ foo: 'bar' }, { baz: 'qux' }]).toIncludeSameMembers([{ foo: 'bar' }, { baz: 'qux' }]);
    expect(new Set([{ foo: 'bar' }, { baz: 'qux' }])).toIncludeSameMembers([{ foo: 'bar' }, { baz: 'qux' }]);
  });

  test('passes when arrays match in a different order', () => {
    expect([1, 2, 3]).toIncludeSameMembers([3, 1, 2]);
    expect(new Set([1, 2, 3])).toIncludeSameMembers([3, 1, 2]);
    expect([{ foo: 'bar' }, { baz: 'qux' }]).toIncludeSameMembers([{ baz: 'qux' }, { foo: 'bar' }]);
  });

  test('fails when the iterables are not equal in length', () => {
    expect(() => expect([1, 2]).toIncludeSameMembers([1])).toThrowErrorMatchingSnapshot();
    expect(() => expect(new Set([1, 2])).toIncludeSameMembers([1, 2, 2])).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toIncludeSameMembers', () => {
  test('fails when array contents match', () => {
    expect(() => expect([1]).not.toIncludeSameMembers([1])).toThrowErrorMatchingSnapshot();
    expect(() => expect(new Set([1])).not.toIncludeSameMembers([1])).toThrowErrorMatchingSnapshot();
  });

  test('passes when the arrays are different in length', () => {
    expect([1, 2, 3]).not.toIncludeSameMembers([1, 2]);
    expect(new Set([1, 2])).not.toIncludeSameMembers([1, 2, 3]);
    expect([{ foo: 'bar' }, { baz: 'qux' }]).not.toIncludeSameMembers([{ hello: 'world' }]);
  });

  test('passes when given object is not an iterable', () => {
    expect(4).not.toIncludeSameMembers([4, 5, 6]);
  });

  test('passes when iterables do not match', () => {
    expect([]).not.toIncludeSameMembers([1]);
    expect([1]).not.toIncludeSameMembers([1, 1]);
    expect([1, 2]).not.toIncludeSameMembers([1, 2, 2]);
    expect([1, 2, 3]).not.toIncludeSameMembers([2, 3, 4]);

    expect(new Set([])).not.toIncludeSameMembers([1]);
    expect(new Set([1])).not.toIncludeSameMembers([1, 1]);
    expect(new Set([1, 2])).not.toIncludeSameMembers([1, 2, 2]);
    expect(new Set([1, 2, 3])).not.toIncludeSameMembers([2, 3, 4]);
  });
});
