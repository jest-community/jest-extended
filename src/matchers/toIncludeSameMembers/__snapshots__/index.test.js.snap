// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toIncludeSameMembers fails when array contents match 1`] = `
"<dim>expect(</><red>received</><dim>).not.toIncludeSameMembers(</><green>expected</><dim>)</>

Expected list to not exactly match the members of:
  <green>[1]</>
Received:
  <red>[1]</>"
`;

exports[`.not.toIncludeSameMembers fails when array contents match 2`] = `
"<dim>expect(</><red>received</><dim>).not.toIncludeSameMembers(</><green>expected</><dim>)</>

Expected list to not exactly match the members of:
  <green>[1]</>
Received:
  <red>Set {1}</>"
`;

exports[`.toIncludeSameMembers fails when the iterables are not equal in length 1`] = `
"<dim>expect(</><red>received</><dim>).toIncludeSameMembers(</><green>expected</><dim>)</>

Expected list to have the following members and no more:
  <green>[1]</>
Received:
  <red>[1, 2]</>"
`;

exports[`.toIncludeSameMembers fails when the iterables are not equal in length 2`] = `
"<dim>expect(</><red>received</><dim>).toIncludeSameMembers(</><green>expected</><dim>)</>

Expected list to have the following members and no more:
  <green>[1, 2, 2]</>
Received:
  <red>Set {1, 2}</>"
`;
